### PR TITLE
[Dask] Dask Example Tests

### PR DESCRIPTION
**Pull request by amogkam**
Adds tests for the Dask on Ray examples in the docs, both with and without Ray Client.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/util/dask/BUILD [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    tags = ["exclusive", "client"],
+    deps = [":dask_lib"],
+)
+
+# This is a dummy test dependency that causes the above tests to be
+# re-run if any of these files changes.
+py_library(
+    name = "dask_lib",
+    srcs = glob(["**/*.py"], exclude=["tests/*.py"]),
+)
```
Comment (rkooo567) [inline python/ray/util/dask/BUILD:118]: newline

**Review by rkooo567 — APPROVED** [review]
Is moving the test file to util necessary change for the client?

**Comment by amogkam** [comment]
@rkooo567 it's not necessary, but I did it so we can have the unit tests and the example tests in the same BUILD file.

File: python/ray/util/dask/BUILD [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
```
Comment (amogkam) [inline python/ray/util/dask/BUILD:118]:
```suggestion
)

```

File: python/ray/util/dask/examples/dask_ray_shuffle_optimization.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import pandas as pd
+
+# Start Ray.
+# Tip: If connecting to an existing cluster, use ray.init(address="auto").
+ray.init()
+
+# Set the Dask DataFrame optimizer to
+# our custom optimization function, this time using the config setter as a
+# context manager.
+with dask.config.set(dataframe_optimize=dataframe_optimize):
```
Comment (amogkam) [inline python/ray/util/dask/examples/dask_ray_shuffle_optimization.py:15]: I had to remove `scheduler=dask_ray_get` from here to get this working with Ray Client. Is this expected? @rkooo567 @clarkzinzow

Comment (clarkzinzow) [reply python/ray/util/dask/examples/dask_ray_shuffle_optimization.py:15]: @amogkam If `scheduler=ray_dask_get` is removed, this example won't use the Dask-on-Ray scheduler, it'll just run the Dask workload on a local thread pool and the optimization will be a no-op.

How does the example fail when the scheduler is set? That seems like a bug.

**Pull request merged**